### ISSUE TITLE
Paranamer issue if Object uses lambda

### ISSUE THREAD

**Issue by muminc** [orika-mapper/orika#60]
FYI,

Came across an issue when trying to do mapping on Objects which uses Java 8 lambdas,  I get exceptions like below.

```

Caused by: java.lang.ArrayIndexOutOfBoundsException: 24950
	at com.thoughtworks.paranamer.BytecodeReadingParanamer$ClassReader.accept(BytecodeReadingParanamer.java:563)
	at com.thoughtworks.paranamer.BytecodeReadingParanamer$ClassReader.access$200(BytecodeReadingParanamer.java:338)
	at com.thoughtworks.paranamer.BytecodeReadingParanamer.lookupParameterNames(BytecodeReadingParanamer.java:103)
	at com.thoughtworks.paranamer.AdaptiveParanamer.lookupParameterNames(AdaptiveParanamer.java:75)
	at com.thoughtworks.paranamer.CachingParanamer.lookupParameterNames(CachingParanamer.java:90)
	at com.thoughtworks.paranamer.CachingParanamer.lookupParameterNames(CachingParanamer.java:83)
	at ma.glasnost.orika.constructor.SimpleConstructorResolverStrategy.resolve(SimpleConstructorResolverStrategy.java:121)
	at ma.glasnost.orika.impl.generator.ObjectFactoryGenerator.addSourceClassConstructor(ObjectFactoryGenerator.java:155)
	at ma.glasnost.orika.impl.generator.ObjectFactoryGenerator.addCreateMethod(ObjectFactoryGenerator.java:123)
	at ma.glasnost.orika.impl.generator.ObjectFactoryGenerator.build(ObjectFactoryGenerator.java:94)

```

Someone already raised a ticket raised for paranamer  https://github.com/paul-hammant/paranamer/issues/17 

I've tested the proposed fix for paranamer, can confirm mapping issue gets fixed for me. So I think it's worthwhile tracking paranamer for the latest release.

**Comment by elaatifi** [comment]
Awesome ! Thank you @muminc

**Comment by elaatifi** [comment]
*UPDATE*: Should be fixed in **1.4.6**
